client/src/related/comparisonModal.jsx

```diff
@@ -0,0 +1,93 @@
+import React from 'react';
+class ComparisonModal extends React.Component {
+  constructor(props) {
+    super(props);
+  }
+
+  render() {
+    if (!this.props.showModal) {
+      return null;
+    } else {
+      var current = this.props.currentProd;
+      var comparison = this.props.comparisonProd;
+      //create an object containing all features present in either product
+      var allFeatures = {};
+      current.features.forEach((feature) => {
+        var featureValue;
+        if (feature.value === true) {
+          featureValue = '✓';
+        } else {
+          featureValue = feature.value;
+        }
+
+        allFeatures[feature.feature] =
+        {
+          feature: feature.feature,
+          current: featureValue
+        };
+      });
+
+      comparison.features.forEach((feature) => {
```

Comment:
Is line 30-36 needed? Looks like a earlier version of 15-28

Reply:
It is indeed needed! This is for the comparison modal which is a table comparing a product from the related list to the current product on the page. So Lines 15-28 pull in the data for the current product on the page, and lines 30-36 pull in the the info for the product that's being compared. That does tell me that I could probably add some comments/do better variable naming next time though to make it more obvious!

Reply:
No, you are fine, lol.  Comparison and current are two different variables, I just can't read.